runtime/common/src/slots.rs

```diff
@@ -427,20 +431,34 @@ impl<T: Config> Leaser for Pallet<T> {
 			.unwrap_or_else(Zero::zero)
 	}
 
-	fn lease_period() -> Self::LeasePeriod {
-		T::LeasePeriod::get()
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	fn lease_period_length() -> (T::BlockNumber, T::BlockNumber) {
+		(T::LeasePeriod::get(), T::LeaseOffset::get())
 	}
 
-	fn lease_period_index() -> Self::LeasePeriod {
-		<frame_system::Pallet<T>>::block_number() / T::LeasePeriod::get()
+	fn lease_period_index(b: T::BlockNumber) -> (Self::LeasePeriod, bool) {
+		// Note that lease lease period 0 is artificially extended by the lease offset.
+		let offset_block_now = b.saturating_sub(T::LeaseOffset::get());
```

Comment:
See previous comment. I would have expected this to be:

```suggestion
		let offset_block_now = b.checked_sub(T::LeaseOffset::get())?;
```